src/main/java/seedu/address/model/customer/Customer.java

```diff
@@ -22,34 +21,24 @@ public class Customer {
     private final Email email;
 
     // Data fields
-    private final Set<Tag> tags = new HashSet<>();
+    private final Set<Tag> tags;
+
+    private final Set<Commission> commissions;
 
     // Optional fields
     private final Address address;
 
     /**
-     * Every field must be present and not null.
-     */
-    public Customer(Name name, Phone phone, Email email, Set<Tag> tags) {
-        requireAllNonNull(name, phone, email, tags);
-        this.name = name;
-        this.phone = phone;
-        this.email = email;
-        this.address = null;
-        this.tags.addAll(tags);
-    }
-
-
-    /**
-     * Every field must be present and not null.
+     * Constructs a Customer.
+     * @param builder Instance of CustomerBuilder.
      */
-    public Customer(Name name, Phone phone, Email email, Address address, Set<Tag> tags) {
-        requireAllNonNull(name, phone, email, address, tags);
-        this.name = name;
-        this.phone = phone;
-        this.email = email;
-        this.address = address;
-        this.tags.addAll(tags);
+    public Customer(CustomerBuilder builder) {
+        this.name = builder.name;
+        this.phone = builder.phone;
+        this.email = builder.email;
+        this.tags = builder.tags;
+        this.commissions = builder.commissions;
+        this.address = builder.address;
```

Comment:
Small nitpick, if I'm not wrong, since we replaced the params with a `CustomerBuilder`, the customer attributes are no longer shadowed so `this` here becomes unnecessary. Maybe let's remove them so it adheres to the [style guide](https://se-education.org/guides/conventions/java/index.html#variables)?

```suggestion
        name = builder.name;
        phone = builder.phone;
        email = builder.email;
        tags = builder.tags;
        commissions = builder.commissions;
        address = builder.address;
```